File: python/programmers/dfsbfs/p1.py
class Number:
    def __init__(self, numbers, target):
        self.numbers = numbers
        self.target = target
        self.n = len(numbers)
        self.answer = 0

    def dfs(self, idx, value):
        if (idx == self.n - 1):
            if value == self.target:
                self.answer += 1
            return
        self.dfs(idx + 1, value + self.numbers[idx + 1])
        self.dfs(idx + 1, value - self.numbers[idx + 1])


def solution(numbers, value):
    n = Number(numbers, value)
    n.dfs(0, numbers[0])
    n.dfs(0, -numbers[0])
    return n.answer

File: java/programmers/src/dfsbfs/p1.java
package dfsbfs;

public class p1 {
	static int n;
	static int answer;
	static int[] numbers;
	static int target;

	static void dfs(int idx, int value) {
			if (idx==n-1) {
				if(value == target) {
					answer+=1;
				}
				return;
			}
			
			dfs(idx+1, value+numbers[idx+1]);
			dfs(idx+1,value-numbers[idx+1]);
			
			
		}

	public static void main(String[] args) {

		numbers = new int[] { 1, 1, 1, 1, 1 };
		target = 3;
		n = numbers.length;

		dfs(0, numbers[0]);
		dfs(0, -numbers[0]);

		System.out.println(answer);

	}

}
